tools/scripts/build/ensure-env.js

```diff
@@ -48,28 +54,47 @@ if (FREECODECAMP_NODE_ENV !== 'development') {
     searchKeys,
     donationKeys
   );
-  const variables = Object.keys(env);
+  const receivedvariables = Object.keys(env);
   expectedVariables.sort();
-  variables.sort();
-  if (expectedVariables.length !== variables.length) {
-    throw Error(`Env. variable validation failed. Expected
-    ${expectedVariables}
-    but recieved
-    ${variables}
+  receivedvariables.sort();
+  if (expectedVariables.length !== receivedvariables.length) {
+    throw Error(`
+
+    Env. variable validation failed.
+
+    These keys are missing:${expectedVariables
+      .filter(expected => !receivedvariables.includes(expected))
+      .concat(
+        receivedvariables.filter(
+          received => !expectedVariables.includes(received)
+        )
+      )}
```

Comment:
If I am reading this correctly, if I have an extra key in my `.env` that isn't expected, the console would throw an error and claim that extra key is "missing"? Is that accurate?

Reply:
Actually, this test is about keys that maybe extraneous in `config/env.json`. They are validated here, so they are "missing" validation (not that they are missing in config, that test comes [little later](https://github.com/freeCodeCamp/freeCodeCamp/pull/40813/files#diff-13b76f3541861e5aa188357796687d0af88adf112242c6b7ecdcca663a0f7c59R79))

I can see how the phraseology here is not clear. In any case these tests were meant for the CD and you would not be affected as a dev. 

I will make the messages more clear.